dist_run.py

```diff
@@ -167,6 +258,17 @@ def main():
     pp_mesh = device_mesh["pp"]
     tp_rank = tp_mesh.get_local_rank()
     pp_rank = pp_mesh.get_local_rank()
+    tp_group = tp_mesh.get_group()
+    pp_group = pp_mesh.get_group()
+
+    logger.info(f"review: {pp_group=}, {tp_group= }")
+
+    logger.info(f"Created device mesh: {device_mesh}\n {tp_mesh=}, {pp_mesh=}\n")
+    # TODO - this assumes 1D mesh, need to update for 2D+ mesh
```

Comment:
why is it a 1D mesh?

Reply:
it's not necessarily, it's more that I could not ensure that the return from .size() is capturing all total if 2D+ and there's no len(device_mesh) to quickly get the total participants.
Thus just marked it as a TODO and have been checking with Iris for a better way.